README.md

```diff
@@ -1,2 +1,135 @@
-# node-firefox-forward-ports
-Forward local ports to remote debugging interfaces on connected Firefox OS devices
+# node-firefox-forward-ports [![Build Status](https://secure.travis-ci.org/mozilla/node-firefox-forward-ports.png?branch=master)](http://travis-ci.org/mozilla/node-firefox-forward-ports)
+
+> Forward local ports to remote debugging interfaces on connected Firefox OS devices
+
+[![Install with NPM](https://nodei.co/npm/node-firefox-forward-ports.png?downloads=true&stars=true)](https://nodei.co/npm/node-firefox-forward-ports/)
+
+This is part of the [node-firefox](https://github.com/mozilla/node-firefox) project.
+
+Firefox OS devices offer a [Remote Debugging service](https://wiki.mozilla.org/Remote_Debugging_Protocol) 
+on a local network socket. This can be forwarded to a host computer over USB
+using ADB. This module takes care of that as necessary, reusing forwarded ports
+when possible.
+
+## Installation
+
+### From git
+
+```bash
+git clone https://github.com/mozilla/node-firefox-forward-ports.git
+cd node-firefox-forward-ports
+npm install
+```
+
+If you want to update later on:
+
+```bash
+cd node-firefox-forward-ports
+git pull origin master
+npm install
+```
+
+### npm
+
+```bash
+npm install node-firefox-forward-ports
+```
+
+## Usage
+
+```javascript
+var forwardPorts = require('node-firefox-forward-ports');
+
+// Forwards ports as necessary, while reusing existing ports.
+// Returns the list of devices annotated with details on forwarded ports.
+forwardPorts({
+  devices: [
```

Comment:
I actually like that idea.

And `devices` is an array, right? No reason you couldn't support both APIs; if it's an array then treat it like the new API, if it's a key/value set use the current API.

I propose that!